lambdas/data-migration1/tests/test-rules.js

```diff
@@ -240,29 +254,105 @@ test.serial('migrateRuleRecord handles nullable fields on source rule data', asy
+  const fakeRule = generateFakeRule({
+    collection: {
+      name: fakeCollection.name,
+      version: fakeCollection.version,
+    },
+    provider: fakeProvider.id,
+    updatedAt: Date.now(),
+  });
 
   await migrateFakeCollectionRecord(fakeCollection, knex);
   await migrateFakeProviderRecord(fakeProvider, knex);
   await migrateRuleRecord(fakeRule, knex);
+
+  const olderFakeRule = {
+    ...fakeRule,
+    updatedAt: Date.now() - 1000,
+  };
+
   await t.throwsAsync(
-    migrateRuleRecord(fakeRule, t.context.knex),
+    migrateRuleRecord(olderFakeRule, t.context.knex),
     { instanceOf: RecordAlreadyMigrated }
   );
 });
 
+test.serial('migrateRuleRecord updates an already migrated record if the updated date is newer', async (t) => {
+  const { knex, fakeCollection, fakeProvider } = t.context;
+  const fakeRule = generateFakeRule({
+    collection: {
+      name: fakeCollection.name,
+      version: fakeCollection.version,
+    },
+    provider: fakeProvider.id,
+    updatedAt: Date.now() - 1000,
+  });
+
+  await migrateFakeCollectionRecord(fakeCollection, knex);
+  await migrateFakeProviderRecord(fakeProvider, knex);
+  await migrateRuleRecord(fakeRule, knex);
+
+  const newerFakeRule = generateFakeRule({
+    ...fakeRule,
+    updatedAt: Date.now(),
+  });
+  await migrateRuleRecord(newerFakeRule, knex);
+
+  const createdRecord = await t.context.knex.queryBuilder()
+    .select()
+    .table('rules')
+    .where({ name: fakeRule.name })
+    .first();
+
+  t.deepEqual(
+    omit(createdRecord, ['cumulus_id', 'collection_cumulus_id', 'provider_cumulus_id']),
+    omit(
+      {
+        name: fakeRule.name,
+        workflow: fakeRule.workflow,
+        meta: fakeRule.meta,
+        arn: fakeRule.rule.arn,
+        type: fakeRule.rule.type,
+        value: fakeRule.rule.value,
+        enabled: true,
+        log_event_arn: fakeRule.rule.logEventArn,
+        execution_name_prefix: fakeRule.executionNamePrefix,
+        payload: fakeRule.payload,
+        queue_url: fakeRule.queueUrl,
+        tags: fakeRule.tags,
+        created_at: new Date(fakeRule.createdAt),
+        updated_at: new Date(newerFakeRule.updatedAt),
+      },
+      ruleOmitList
+    )
+  );
+});
+
 test.serial('migrateRules skips already migrated record', async (t) => {
   const { knex, fakeCollection, fakeProvider } = t.context;
-  const fakeRule = generateFakeRule(fakeCollection.name, fakeCollection.version, fakeProvider.id);
+  const fakeRule = generateFakeRule({
+    collection: {
+      name: fakeCollection.name,
+      version: fakeCollection.version,
+    },
+    provider: fakeProvider.id,
+  });
   const queueUrls = randomString();
   fakeRule.queueUrl = queueUrls.queueUrl;
 
+  // This always sets updatedAt to Date.now()
+  await rulesModel.create(fakeRule);
+
+  // We need to make the updateAt of the record we're about to migrate later
+  // than the record in the dynamo table.
+  fakeRule.updatedAt = Date.now();
```

Comment:
I didn't want to change the Dynamo model for the sake of this test. The Rule Dynamo model implements its own version of `create` which sets the `updated_at` to `Date.now()`. That just means that I can't give it a different date when we create the Dynamo record.

Reply:
💥 I think @markdboyd handled this in https://github.com/nasa/cumulus/pull/2164/files?file-filters%5B%5D=.js

I will adjust here.

Reply:
Yeah I was just gonna say that. But what are you planning to change here? Would it be easier to wait until my PR's changes are merged?

Reply:
That might make sense. It's not a huge deal but I should be able to remove this line and set `updatedAt` when we create the Dynamo record in `await rulesModel.create(fakeRule);`. I'll double-check once your changes are merged though. The above PR will likely get in first anyway.